js/test/hooks/useEmployees.test.ts

```diff
@@ -0,0 +1,60 @@
+import { fetch } from "../../browser";
+import { findEmployeeByBadge, useEmployees } from "../../hooks/useEmployees";
+import { PromiseWithResolvers } from "../helpers/promiseWithResolvers";
+import { renderHook, waitFor } from "@testing-library/react";
+
+jest.mock("../../browser", () => ({
+  __esModule: true,
+  fetch: jest.fn(),
+}));
+
+const TEST_DATA = {
+  data: [
+    {
+      first_name: "Christopher",
+      preferred_first: "Chris",
+      last_name: "Robin",
+      badge: "123",
+    },
+  ],
+};
+
+const TEST_PARSED = [
+  {
+    first_name: "Christopher",
+    preferred_first: undefined,
+    last_name: "Robin",
+    badge: "123",
+  },
+];
+
+describe("useEmployees", () => {
+  test("parses api response", async () => {
+    const { promise, resolve } = PromiseWithResolvers<Response>();
+    jest.mocked(fetch).mockReturnValue(promise);
+
+    const { result } = renderHook(useEmployees);
+
+    resolve({
+      status: 200,
+      json: () =>
+        new Promise((resolve) => {
+          resolve(TEST_DATA);
+        }),
+    } as Response);
+
+    await waitFor(() => {
+      expect(result.current).toEqual({ status: "ok", result: TEST_PARSED });
+    });
```

Comment:
**thought (non-blocking):** I'm actually kind of surprised that neither the `resolve` nor the `waitFor` seem to need to be wrapped in `act` to avoid the tests from throwing an error / warning about React state updates happening outside of `act` or certain other predetermined places.

Reply:
Nah, I don't think that would make a difference, since the issue here is that it happens outside of the usual React render cycle. Anyhow, I'm not going to complain too much if it works.